akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -495,25 +496,33 @@ class CassandraProjectionSpec
 
       val handler = concatHandlerFail4()
 
-      val projection =
+      val statusProbe = createTestProbe[TestStatusObserver.Status]()
+      val statusObserver = new TestStatusObserver[Envelope](statusProbe.ref, lifecycle = true)
+
+      val projectionFailing =
         CassandraProjection
           .atLeastOnce[Long, Envelope](projectionId, sourceProvider(entityId), () => handler)
           .withSaveOffset(2, 1.minute)
           .withRecoveryStrategy(HandlerRecoveryStrategy.retryAndFail(3, 10.millis))
+          .withStatusObserver(statusObserver)
 
-      intercept[TestException] {
-        projectionTestKit.run(projection) {
-          withClue("checking: all expected values were concatenated") {
-            val concatStr = repository.findById(entityId).futureValue.get
-            concatStr.text shouldBe "abc|def|ghi"
-          }
-        }
+      spawn(ProjectionBehavior(projectionFailing))
+      eventually {
+        val concatStr = repository.findById(entityId).futureValue.get
+        concatStr.text shouldBe "abc|def|ghi"
       }
```

Comment:
The concatenated string can be checked against "abc|def|ghi" while the fourth element was not yet processed. 

As a result the callback completes successfully and the projection stops before if gets the chance to fail on the fourth element.